Objective-C/CBLFullTextIndexConfiguration.h

```diff
@@ -41,11 +41,25 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, readonly, nullable) NSString* language;
 
 /**
- Constructor for creating a full-text index by using an array of expression strings
+ A predicate expression defining conditions for indexing documents.
+ Only documents satisfying the predicate are included, enabling partial indexes.
+ */
+@property (nonatomic, readonly, nullable) NSString* where;
+
+/**
+ Initializes a partial full-text index by using an array of expression strings
+ */
+- (instancetype) initWithExpression: (NSArray<NSString*>*)expressions
+                      ignoreAccents: (BOOL)ignoreAccents
+                           language: (nullable NSString*)language;
+
+/**
+ Initializes a partial full-text index with an array of expression strings and a where clause.
  */
 - (instancetype) initWithExpression: (NSArray<NSString*>*)expressions
                       ignoreAccents: (BOOL)ignoreAccents
-                           language: (NSString* __nullable)language;
+                           language: (nullable NSString*)language
+                              where: (nullable NSString*)where;
```

Comment:
Sorry, I didn't notice this in my first review. I feel like where should be next to the expression and that is based on what is in the API spec as well. Can you move where next to the expressions? 

For the comment, feel like mentioning the partial index next to where clause feels better as it's an optional. What do you think?

```
Initializes a full-text index with an array of expression strings and an optional where clause for a partial index.
```